LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/ConcentratorDeduplication.cs

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace LoRaWan.NetworkServer
+{
+    using LoRaWan.NetworkServer.BasicsStation;
+    using Microsoft.Extensions.Caching.Memory;
+    using Microsoft.Extensions.Logging;
+    using System;
+    using System.Security.Cryptography;
+    using System.Text;
+
+    public sealed class ConcentratorDeduplication : IConcentratorDeduplication, IDisposable
```

Comment:
This does de-duplication of uplink data frames rather than _concentrators_ so isn't it better to call it `UplinkDataFrameDeduplicator`?

Reply:
Hm, I see your point. The reasoning was that the criterion for the deduplication is from which concentrator we got it from. This is important in order to differentiate with the other deduplication service that is based on the LNS used to send a message upstream (and utilizes the Azure Function). I think using only `UplinkDataFrame` in the name will not provide enough semantics.

Reply:
Resolving this but feel free to reopen if you want to.